### ISSUE TITLE
No support for Implicit Grant's

### ISSUE THREAD

**Issue by AlastairDewar** [dusterio/lumen-passport#36]
There is no support for implicit grants as the PassportServiceProvider is missing the forAuthorization method which sets up the routing for the authorize endpoints (should be called in registerRoutes).

**Comment by AlastairDewar** [comment]
Some notes on what I've done so far ....

Add the following function after registerRoutes in ./vendor/dusterio/lumen-passport/src/PassportServiceProvider.php
`/**
     * Register the routes needed for authorization.
     *
     * @return void
     */
    public function forAuthorization()
    {
        $this->app->get('/oauth/authorize', [
            'uses' => 'Laravel\Passport\Http\Controllers\AuthorizationController@authorize',
        ]);

        $this->app->group(['middleware' => ['auth']], function () {
            $this->app->post('/oauth/authorize', [
                'uses' => 'Laravel\Passport\Http\Controllers\ApproveAuthorizationController@approve',
            ]);

            $this->app->delete('/oauth/authorize', [
                'uses' => 'Laravel\Passport\Http\Controllers\DenyAuthorizationController@deny',
            ]);
        });
    }`

Add the following line to be the first line in registerRoutes in ./vendor/dusterio/lumen-passport/src/PassportServiceProvider.php
`$this->forAuthorization();`

Make sure the following line is uncommented in ./bootstrap/app.php
`$app->register(App\Providers\AppServiceProvider::class);`

Add the following block to the register method in ./app/Providers/AppServiceProvider.php
`$this->app->singleton('Illuminate\Contracts\Routing\ResponseFactory', function ($app) {
            return new \Illuminate\Routing\ResponseFactory(
                $app['Illuminate\Contracts\View\Factory'],
                $app['Illuminate\Routing\Redirector']
            );
        });`

**Comment by dusterio** [comment]
Doesn't this require sessions? That's why I didn't include it - because Lumen runs without them by default. Also, this is not implicit grant ;-) Implicit grant in OAuth 2 is when you redirect back and pass an access token
This is just a manual way to manage access tokens

**Comment by jake142** [comment]
Trying to figure out which grants are actually available? Is it only password grant?

**Comment by dusterio** [comment]
Grants that don't require session and web views - password grant, refresh grant and client credentials grant. Implicit and auth code grants would have required both sessions and views (pages)

**Comment by jake142** [comment]
Ah, ok. So if I want 3e parties (not trusted) to use my API I need to go for a full Laravel passport implementation. Thanks for the information.

**Comment by dusterio** [comment]
Yep, it's a limitation introduced by Lumen, not by this package :) Having said that, I will consider adding views and missing grant types to Lumen Passport in the future - it kind of makes sense sometimes (eg. you have a OAuth 2 microservice)

**Comment by jake142** [comment]
Agree, I use JWT (tymon) today and the reason I would go for passport would be to expose the API to untrusted 3e party clients.

Do you think it would take a lot of effort to implement the missing grants? Views seems to work in my installation (return view('my.test'))? Would it be possible to use the PHP session for passport?
